fast_barcode_scanner/lib/src/camera_controller.dart

```diff
@@ -139,7 +167,18 @@ class _CameraController implements CameraController {
       state._previewConfig = await _platform.init(
           types, resolution, framerate, detectionMode, position);
 
-      _onScan = onScan;
+      _onScan = (barcode) {
```

Comment:
How does `_scanSilencerSubscription` work? This seems accept multiple barcodes already?

Reply:
Good question. The scan silencer subscription helps us know whether the scanned codes are still visible on the screen.

So long as the barcode is visible, the platform will continuously return the scanned value of the code.

But what happens when you move the camera away from the code? The answer is nothing. The last frame that had the code sent the code back to us, but then the next frame has no code. The ValueNotifier still holds the value of the last scanned code, but no code is still visible.

The solution is to have a slower periodic timer that checks whether we've had any recent scans. If the last scan is recent enough, then we sleep and check later. If the last scan is past the timeout, we decide that the code is no longer visible on the screen and we remove it from the ValueNotifier.

https://user-images.githubusercontent.com/959931/153494139-9c4b6398-9d34-44ac-837d-6a509329dfe9.mp4

https://user-images.githubusercontent.com/959931/153494279-09c323fd-86f3-45d0-80d7-f58a8bb451e0.mp4